File: cakk-api/src/main/java/com/cakk/api/mapper/EventMapper.java
package com.cakk.api.mapper;

import jakarta.servlet.http.HttpServletRequest;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import com.cakk.core.dto.event.CakeIncreaseViewsEvent;
import com.cakk.core.dto.event.CakeShopIncreaseViewsEvent;
import com.cakk.core.dto.event.EmailWithVerificationCodeSendEvent;
import com.cakk.core.dto.event.ErrorAlertEvent;
import com.cakk.core.dto.event.IncreaseSearchCountEvent;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class EventMapper {

	public static EmailWithVerificationCodeSendEvent supplyEmailWithVerificationCodeSendEventBy(final String email, final String code) {
		return new EmailWithVerificationCodeSendEvent(email, code);
	}

	public static IncreaseSearchCountEvent supplyIncreaseSearchCountEventBy(final String keyword) {
		return new IncreaseSearchCountEvent(keyword);
	}

	public static ErrorAlertEvent supplyErrorAlertEventBy(
		final Exception exception,
		final HttpServletRequest request,
		final String profile
	) {
		return new ErrorAlertEvent(exception, request, profile);
	}

	public static CakeIncreaseViewsEvent supplyCakeIncreaseViewsEventBy(Long cakeId) {
		return new CakeIncreaseViewsEvent(cakeId);
	}

	public static CakeShopIncreaseViewsEvent supplyCakeShopIncreaseViewsEventBy(Long cakeShopId) {
		return new CakeShopIncreaseViewsEvent(cakeShopId);
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/mapper/EventMapper.kt
package com.cakk.core.mapper

import com.cakk.core.dto.event.*
import com.cakk.infrastructure.persistence.shop.CertificationEvent
import com.cakk.external.vo.message.CertificationMessage
import com.cakk.external.vo.message.ErrorAlertMessage
import com.cakk.external.vo.message.VerificationMessage
import jakarta.servlet.http.HttpServletRequest

fun supplyCakeIncreaseViewsEventBy(cakeId: Long): CakeIncreaseViewsEvent {
	return CakeIncreaseViewsEvent(cakeId)
}

fun supplyCakeShopIncreaseViewsEventBy(cakeShopId: Long): CakeShopIncreaseViewsEvent {
	return CakeShopIncreaseViewsEvent(cakeShopId)
}

fun supplyErrorAlertEventBy(
	exception: Exception,
	request: HttpServletRequest,
	profile: String
): ErrorAlertEvent {
	return ErrorAlertEvent(exception, request, profile)
}

fun supplyEmailWithVerificationCodeSendEventBy(email: String, code: String): EmailWithVerificationCodeSendEvent {
	return EmailWithVerificationCodeSendEvent(email, code)
}

fun supplyCertificationMessageBy(certificationEvent: com.cakk.infrastructure.persistence.shop.CertificationEvent): CertificationMessage {
	return CertificationMessage(
		certificationEvent.businessRegistrationImageUrl,
		certificationEvent.idCardImageUrl,
		certificationEvent.emergencyContact,
		certificationEvent.message,
		certificationEvent.userId,
		certificationEvent.userEmail,
		certificationEvent.shopName,
		certificationEvent.location.y,
		certificationEvent.location.x
	)
}

fun supplyVerificationMessageBy(event: EmailWithVerificationCodeSendEvent): VerificationMessage {
	return VerificationMessage(event.email, event.code)
}

fun supplyErrorAlertMessageBy(event: ErrorAlertEvent): ErrorAlertMessage {
	val profile: String = event.profile
	val stackTrace: String = event.exception.stackTrace.contentToString()
	val request: HttpServletRequest = event.request

	return ErrorAlertMessage(
		profile,
		stackTrace,
		request.contextPath,
		request.requestURL.toString(),
		request.method,
		request.parameterMap,
		request.remoteAddr,
		request.getHeader("User-Agent")
	)
}

fun supplyIncreaseSearchCountEventBy(keyword: String): IncreaseSearchCountEvent {
	return IncreaseSearchCountEvent(keyword)
}
